### PR TITLE
replace deprecated gtk-image-menu-item*

### PR DESCRIPTION
**Pull request by raveit65**

### CODE REVIEW

File: applets/notification_area/status-notifier/sn-dbus-menu-item.c [hunk 1]
```diff
@@ -203,9 +204,9 @@ sn_dbus_menu_item_new (GVariant *props)
               cairo_surface_destroy (surface);
             }
 
-          item->item = gtk_image_menu_item_new ();
+          item->item = mate_image_menu_item_new ();
           gtk_image_menu_item_set_always_show_image (GTK_IMAGE_MENU_ITEM (item->item), TRUE);
```
Comment (zhuyaliang) [inline applets/notification_area/status-notifier/sn-dbus-menu-item.c:208]: If ```mate_image_menu_item_set_image```  is used, there is no need to use ```gtk_image_menu_item_set_always_show_image```again

File: mate-panel/menu.c [hunk 2]
```diff
@@ -964,7 +965,7 @@ panel_image_menu_item_new (void)
 {
 	GtkWidget *menuitem;
 
-	menuitem = gtk_image_menu_item_new ();
+	menuitem = mate_image_menu_item_new ();
 	gtk_image_menu_item_set_always_show_image (GTK_IMAGE_MENU_ITEM (menuitem),
```
Comment (zhuyaliang) [inline mate-panel/menu.c:969]: Remove this line

**Review by lukefromdc — APPROVED** [review]
This builds and runs without issue here-and the menus-have-icons gsettings preference is still respected.

Just a straightforward replacement of deprecated gtk_image_menu_item with our own mate_image_menu_item , note that if for any reason we are ever forced to deal with GTK4 this and similar PR's elsewhere will restore what would otherwise be a lost feature. In the meantime by maintaining it ourselves we ensure it does not bitrot

**Review by lukefromdc — COMMENTED** [review]
Update: on running mate-panel from terminal with this applied, I just got many iterations of these warnings:

```
gtk_image_menu_item_set_always_show_image: assertion 'GTK_IS_IMAGE_MENU_ITEM (image_menu_item)' failed

(mate-panel:46673): GLib-GObject-CRITICAL **: 19:07:45.983: invalid cast from 'MateImageMenuItem' to 'GtkImageMenuItem'
```

looks like we need to remove `gtk_image_menu_item_set_always_show_image` and when that is removed from menu.c at line 969 and from sn-dbus-menu-item.c at line 208 the terminal warnings are stopped

File: applets/notification_area/status-notifier/sn-dbus-menu-item.c (hunk 1 above)
Comment (lukefromdc) [reply applets/notification_area/status-notifier/sn-dbus-menu-item.c:208]: Two issues here: using `gtk_image_menu_item_set_always_show_image` with mate_image_menu_item is (presumably) invalid and throws runtime warnings when the panel is run from terminal.

Also note that we have
` if (GTK_IS_IMAGE_MENU_ITEM (item->item))`
at line 441, I would expect it to always test FALSE

though for some reason I still got indicator menu item icons

File: mate-panel/menu.c (hunk 2 above)
Comment (raveit65) [reply mate-panel/menu.c]: done.

File: applets/notification_area/status-notifier/sn-dbus-menu-item.c (hunk 1 above)
Comment (raveit65) [reply applets/notification_area/status-notifier/sn-dbus-menu-item.c]: Looks like i need to change `if (GTK_IS_IMAGE_MENU_ITEM (item->item))` to `if (MATE_IS_IMAGE_MENU_ITEM (item->item))` at line 441 and 450?

**Comment by raveit65** [comment]
With last push i removed `gtk_image_menu_item_set_always_show_image` and runtime warnings are gone.

**Review by zhuyaliang — APPROVED** [review]

File: applets/notification_area/status-notifier/sn-dbus-menu-item.c (hunk 1 above)
Comment (raveit65) [reply applets/notification_area/status-notifier/sn-dbus-menu-item.c]: @zhuyaliang see previous post ^^^^

Comment (raveit65) [reply applets/notification_area/status-notifier/sn-dbus-menu-item.c]: sorry, browser cache wasn't updated.

Comment (raveit65) [reply applets/notification_area/status-notifier/sn-dbus-menu-item.c]: Using `if (MATE_IS_IMAGE_MENU_ITEM (item->item))` with last push.

**Comment by lukefromdc** [comment]
the `GTK_IS_IMAGE_MENU_ITEM` macro seems to work fine, and we already use it in mate-desktop. With this in use the runtime warnings are gone and the entire status notfier built with no warnings at all with my usual build defaults.

Looks good to go here

**Comment by lukefromdc** [comment]
Turns out we lost the Places and System icons in the main menus and had a couple leftover deprecation warnings about `gtk_image_menu_item` 
That came from still defining these two menu items with GtkImageMenuItem and was easily fixed:

https://github.com/mate-desktop/mate-panel/pull/1410

**Comment by raveit65** [comment]
Opps, i am using the classic panel menu in result i didn't saw that.
Thanks.

**Pull request merged**